go/teams/loader.go

```diff
@@ -1402,3 +1408,61 @@ func (l *TeamLoader) NotifyTeamRename(ctx context.Context, id keybase1.TeamID, n
 
 	return nil
 }
+
+func (l *TeamLoader) getHeadMerkleSeqno(mctx libkb.MetaContext, readSubteamID keybase1.TeamID, state *keybase1.TeamSigChainState) (ret keybase1.Seqno, err error) {
+	defer mctx.CTrace("TeamLoader#getHeadMerkleSeqno", func() error { return err })()
+
+	if state.HeadMerkle != nil {
+		return state.HeadMerkle.Seqno, nil
+	}
+	headSeqno := keybase1.Seqno(1)
+	expectedLinkRaw, ok := state.LinkIDs[headSeqno]
+	if !ok {
+		return ret, fmt.Errorf("couldn't find head link in team state during audit")
+	}
+	expectedLink, err := libkb.ImportLinkID(expectedLinkRaw)
+	if err != nil {
+		return ret, err
+	}
+	teamUpdate, err := l.world.getLinksFromServer(mctx.Ctx(), state.Id, []keybase1.Seqno{headSeqno}, &readSubteamID)
+	if err != nil {
+		return ret, err
+	}
+	newLinks, err := teamUpdate.unpackLinks(mctx.Ctx())
+	if err != nil {
+		return ret, err
+	}
+	if len(newLinks) != 1 {
+		return ret, fmt.Errorf("expected only one chainlink back; got %d", len(newLinks))
+	}
+	headLink := newLinks[0]
+	if headLink.Seqno() != headSeqno {
+		return ret, NewInvalidLink(headLink, "wrong head seqno; wanted 1 but got something else")
+	}
+	if !headLink.LinkID().Eq(expectedLink) {
+		return ret, NewInvalidLink(headLink, "wrong head link hash: %s != %s", headLink.LinkID, expectedLink)
+	}
+	if headLink.isStubbed() {
+		return ret, NewInvalidLink(headLink, "got a stubbed head link, but wasn't expecting that")
+	}
+	headMerkle := headLink.inner.Body.MerkleRoot.ToMerkleRootV2()
+	state.HeadMerkle = &headMerkle
+	return headMerkle.Seqno, nil
```

Comment:
Does anything here check that headLink.inner matches headLink.outer?

Reply:
ah good point, i just assumed the unpack function did that, but maybe that was a bad assumption

Reply:
Great catch on that one!